File: task03/task03.go
package main

// fmt package provides the function to print anything
import "fmt"

func printPrimeNumbersBeforeN(N int) {
   // running the for loop from 1 to N
   for i := 2; i <= N; i++ {

      // flag which will confirm that the number is Prime or not
      isPrime := true
      
   // This for loop is checking that the current number have
   // any divisible other than 1
   for j := 2; j <= i/2; j++ {
      if i%j == 0 {
         isPrime = false
         break
      }
   }
   // if the value of the flag is false then the number is not prime
   // and we are not printing it.
   if isPrime {
         fmt.Println(i)
      }
   }
}
func main() {
   // declaring the variable N till which we have to find the Prime Numbers
   var N int
   fmt.Println("Enter the value of N.")  
   // Taking the input of N from the user
   fmt.Scanln(&N)   
   fmt.Println("The prime numbers between 1 to", N, "where N is included are  -")
   
   // calling the function to find and print the prime numbers printPrimeNumbersBeforeN(N)
}

File: task03/task03.rs
// Rust program to print prime numbers 
// from the array

fn main() 
{
    let mut arr:[usize;5] = [1,5,11,26,23];
    let mut flag:i32 = 0;
    let mut i:usize = 0;
    let mut j:usize = 0;
	
    println!("Prime Numbers: ");
    while i<arr.len()
    {
    	flag = 0;
    	j=2;
    	while j<(arr[i]/2)
    	{
    		if arr[i]%j == 0 {
    			flag = 1;
    			break;
    		}
    		j=j+1;
    	}
    	if (flag == 0) && (arr[i]>1) {
    		print!("{} ", arr[i]);
    	}
    	i=i+1;
    }
}
